mediapipe/examples/android/src/java/com/google/mediapipe/apps/instantmotiontracking/MainActivity.java

```diff
@@ -529,23 +555,22 @@ private void prepareDemoAssets() {
   }
 
   private class MediaPipePacketManager implements FrameProcessor.OnWillAddFrameListener {
-    int gif_idx = 0;
     @Override
     public void onWillAddFrame(long timestamp) {
-      Bitmap bmp = defaultGIFTexture;
-      // Cycle through every possible frame
-      if(gif_idx >= GIFBitmaps.size())
-        gif_idx = 0;
-      // If GIF frames are available
-      if(GIFBitmaps.size() > 0)
-        bmp = GIFBitmaps.get(gif_idx++);
+      // set current GIF bitmap as default texture
+      Bitmap currentGIFBitmap = defaultGIFTexture;
+      // If current index is in bounds, display current frame
+      if(gifCurrentIndex <= GIFBitmaps.size() - 1)
```

Comment:
can this if statement ever fail?  if not, let's remove.  Or if it reduces to "if (GIFBitmaps.size() > 0)" then use that instead.

Reply:
Just to clarify: does that mean this if-statement was actually necessary?  Or can we remove it?

Reply:
Not necessary, the logic that you had mentioned previously fixed the issue. So I went ahead and removed the if-statement altogether.